jepsen/test/jepsen/control_test.clj

```diff
@@ -1,8 +1,27 @@
 (ns jepsen.control-test
   (:require [jepsen.control :as c]
-            [clojure.test :refer :all]))
+            [slingshot.slingshot :refer [try+ throw+]]
+            [clojure.test :refer :all]
+            [jepsen.util :refer [contains-many?]]))
 
-(deftest ^:integration basic-test
-  (c/with-ssh {}
-    (c/on "n1"
-          (is (= (c/exec :whoami) "root")))))
+(deftest ^:integration session-test
+  (testing "on failure, session throws debug data"
+    (try+
+     (c/with-ssh {}
+       (c/on "thishostshouldnotresolve"
+             (c/exec :echo "hello")))
```

Comment:
Yeah this is absolutely possible! The keys for the thrown map are `:exit :action :host :out :err :in` because I merged in the response from `jepsen.control/ssh*` to keep it simple. Changing the keys, including using namespaced keys would be straightforward for me to do.

Reply:
Oh that makes perfect sense! Maybe add a :type field too--that's the only one that really needs a namespaced ID I think. :)

Reply:
Awesome, looks good!